### PR TITLE
Respond with the Bot's capabilities #15

### PR DESCRIPTION
**Pull request by jesseolsen**
Possible Inputs:

```
help
what can you do (for me)
```

Output:

```
I can do lots of things.  I can:
Server Profile commands:
  Show all (server) profiles.
  Show a specific (server) profile.
  Show a specific (server) profile compliance.
  Make a specific (server) profile compliant (or update its compliance).
  Power on/off a specific (server) profile.
Server Profile Template commands:
  Show all (server) profile templates.
  Show available targets for a server profile template.
  Show profile(s) using a server profile template.
  Create profile(s) using a server profile template.
  Flex/grow a server profile template by a given amount.
  Remove a number of profiles/servers from a profile template.
  Fix compliance for a profile template.
Server Hardware commands:
  Power on/off a specific (server) hardware.
  List all (server) hardware.
  List a server hardware utilization.
  List a server profile utilization.
Just ask.
```

### CODE REVIEW

**Comment by jesseolsen** [comment]
Some more changes to the PR...

**Comment by gregfuruiye** [comment]
@jesseolsen, here's some things @danielreed and I were discussing a ways back. My apologies for not responding sooner.

It seems appropriate to have couple kinds of responses that might help guide the user to more useful information more quickly. I like the idea of using a "Here's what I can do" kind of thing. We can make this interesting when the Bot has a final personality we can use to create the language for this, for now, here's my quick thoughts:

user: help
hubot: What can I help you with today? (it would be cool if the bot can take context from what the user had just been working on to provide possible solutions, i.e. hubot: I can help you do lots of things with server profiles, such as...)

user: sh
hubot: I can help do a lot of things with server hardware, such as...

user: how do i see bad profiles
hubot: I can list profiles that have error alerts. Would you like to see all of them, or just the ones in a particular enclosure?
user: all
hubot: I'm on it. This might take a while, one sec...

You get the idea.

**Comment by jesseolsen** [comment]
*** Updated PR... Here's the new dialogue... ***
bot> @bot help
I can do lots of things.  Try:
 * Server Profile (sp) help
 * Server Profile Template (spt) help
 * Server hardware (sh) help
Just ask.

bot> @bot sp help
Server Profile (sp) commands:
  Show all (server) profiles (e.g. show all profiles).
  Show a specific (server) profile (e.g. show 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
  Show a specific (server) profile compliance (e.g. show 12325dd7-2108-481a-b1ef-4bfa4e69dabc compliance).
  Make/update a specific (server) profile compliance (e.g. make 12325dd7-2108-481a-b1ef-4bfa4e69dabc compliant).
  Power on/off a specific (server) profile (e.g. turn on 12325dd7-2108-481a-b1ef-4bfa4e69dabc).).

bot> @bot xyz help
I can do lots of things.  Try:
 * Server Profile (sp) help
 * Server Profile Template (spt) help
 * Server hardware (sh) help
Just ask.

bot> @bot all help
I can do lots of things.  I can:
Server Profile (sp) commands:
  Show all (server) profiles (e.g. show all profiles).
  Show a specific (server) profile (e.g. show 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
  Show a specific (server) profile compliance (e.g. show 12325dd7-2108-481a-b1ef-4bfa4e69dabc compliance).
  Make/update a specific (server) profile compliance (e.g. make 12325dd7-2108-481a-b1ef-4bfa4e69dabc compliant).
  Power on/off a specific (server) profile (e.g. turn on 12325dd7-2108-481a-b1ef-4bfa4e69dabc).)
Server Profile Template (spt) commands:
  Show all (server) profile templates (e.g. show all templates).
  Show available targets for a server profile template (e.g. show available targets for 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
  Show profile(s) using a server profile template (e.g. show profile using 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
  Create profile(s) using a server profile template (e.g. create profile for 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
  Flex/grow a server profile template by a given amount (e.g. grow 12325dd7-2108-481a-b1ef-4bfa4e69dabc by 4 profiles).
  Remove a number of profiles/servers from a profile template (e.g. remove 2 profiles from 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
  Fix compliance for a profile template (e.g. fix compliance for 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
Server hardware (sh) commands:
  Power on/off a specific (server) hardware (e.g. turn on 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
  List all (server) hardware (e.g. list all hardware).
  List a server hardware utilization (e.g. list 12325dd7-2108-481a-b1ef-4bfa4e69dabc utilization).
  List a server profile (e.g. list 12325dd7-2108-481a-b1ef-4bfa4e69dabc).
Just ask.

File: src/listener/bot.js [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  ListActions(msg) {
+    return this.transform.text(msg, this.GetActions());
+  }
+
+  GetActions() {
+    return "What can I help you with today? Here's just a few things I can do:\n" +
+      BULLET + this.serverProfiles.title + " help\n" +
+      BULLET + this.serverProfileTemplate.title + " help\n" +
+      BULLET + this.serverHardware.title + " help\n" +
+//      BULLET + this.developer.capabilitiesHeader + "\n" +
```
Comment (danielreed) [inline src/listener/bot.js]: Did you mean for this to be commented out?

**Pull request merged**